pl_bolts/models/detection/components/torchvision_backbones.py

```diff
@@ -0,0 +1,102 @@
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+# Use this when you have Adaptive Pooling layer in End.
+# When Model.features is not applicable.
+def _create_backbone_adaptive(model, out_channels: int = None):
+    """
+    Creates backbone by removing linear after Adaptive Pooling layer.
+    Args:
+        model: torch.nn model with adaptive pooling layer.
+        out_channels (Optional) : Number of out_channels in last layer.
+    """
+    if out_channels is None:
+        modules_total = list(model.children())
+        out_channels = modules_total[-1].in_features
+    return _create_backbone_generic(model, out_channels=out_channels)
+
+
+def _create_backbone_features(model, out_channels: int):
+    """
+    Creates backbone from feature sequential block.
+    Args:
+        model: torch.nn model with features as sequential block.
+        out_channels: Number of out_channels in last layer.
+    """
+    ft_backbone = model.features
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+def create_torchvision_backbone(model_name: str, pretrained: bool = True):
+    """
+    Creates CNN backbone from Torchvision.
+    Args:
+        model_name (str) : Name of the model. E.g. resnet18
+        pretrained (str) : Pretrained weights dataset "imagenet", etc
+    """
+
+    if model_name == "mobilenet_v2":
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        out_channels = 1280
+        ft_backbone = _create_backbone_features(net, 1280)
+        return ft_backbone, out_channels
+
+    elif model_name in ["vgg11", "vgg13", "vgg16", "vgg19", ]:
+        out_channels = 512
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        ft_backbone = _create_backbone_features(net, out_channels)
+        return ft_backbone, out_channels
+
+    elif model_name in ["resnet18", "resnet34"]:
+        out_channels = 512
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        ft_backbone = _create_backbone_adaptive(net, out_channels)
+        return ft_backbone, out_channels
+
+    elif model_name in ["resnet50", "resnet101", "resnet152", "resnext50_32x4d", "resnext101_32x8d", ]:
+        out_channels = 2048
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        ft_backbone = _create_backbone_adaptive(net, 2048)
+        return ft_backbone, out_channels
+
+    elif model_name in ["mnasnet0_5", "mnasnet0_75", "mnasnet1_0", "mnasnet1_3"]:
+        out_channels = 1280
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        ft_backbone = _create_backbone_adaptive(net, 1280)
+        return ft_backbone, out_channels
+
+    else:
+        raise ValueError("Unsupported model")
```

Comment:
Maybe echo back model_name?
```suggestion
        raise ValueError(f"Unsupported model: '{model_name}'")
```